packages/celotool/src/lib/oracle.ts

```diff
@@ -37,86 +107,167 @@ async function helmParameters(celoEnv: string) {
+  ].concat(await oracleIdentityHelmParameters(context, oracleConfig))
 }
 
-async function oracleIdentityHelmParameters(celoEnv: string) {
-  const addressesAndVaults = oracleAddressesAndVaults()
-  const replicas = addressesAndVaults.length
+async function oracleIdentityHelmParameters(
+  context: OracleAzureContext,
+  oracleConfig: OracleConfig
+) {
+  const replicas = oracleConfig.identities.length
   let params: string[] = []
   for (let i = 0; i < replicas; i++) {
-    const identity = await createOracleIdentityIfNotExists(celoEnv, i)
-    const { address, keyVaultName: vaultName } = addressesAndVaults[i]
+    const oracleIdentity = oracleConfig.identities[i]
+    const azureIdentity = await createOracleAzureIdentityIfNotExists(context, oracleIdentity)
     const prefix = `--set oracle.identities[${i}]`
     params = params.concat([
-      `${prefix}.address=${address}`,
-      `${prefix}.azure.id=${identity.id}`,
-      `${prefix}.azure.clientId=${identity.clientId}`,
-      `${prefix}.azure.keyVaultName=${vaultName}`,
+      `${prefix}.address=${oracleIdentity.address}`,
+      `${prefix}.azure.id=${azureIdentity.id}`,
+      `${prefix}.azure.clientId=${azureIdentity.clientId}`,
+      `${prefix}.azure.keyVaultName=${oracleIdentity.keyVaultName}`,
     ])
   }
   return params
 }
 
-function releaseName(celoEnv: string) {
-  return `${celoEnv}-oracle`
-}
-
-async function createOracleIdentityIfNotExists(celoEnv: string, index: number) {
-  const identity = await createIdentityIfNotExists(oracleIdentityName(celoEnv, index))
+async function createOracleAzureIdentityIfNotExists(
+  context: OracleAzureContext,
+  oracleIdentity: OracleIdentity
+) {
+  const clusterConfig = getAzureClusterConfig(context)
+  const identity = await createIdentityIfNotExists(clusterConfig, oracleIdentity.azureIdentityName)
 
   // Grant the service principal permission to manage the oracle identity.
   // See: https://github.com/Azure/aad-pod-identity#6-set-permissions-for-mic
   const [rawServicePrincipalClientId] = await execCmdWithExitOnFailure(
-    `az aks show -n ${clusterName()} --query servicePrincipalProfile.clientId -g ${resourceGroup()} -o tsv`
+    `az aks show -n ${clusterConfig.clusterName} --query servicePrincipalProfile.clientId -g ${clusterConfig.resourceGroup} -o tsv`
   )
   const servicePrincipalClientId = rawServicePrincipalClientId.trim()
   await execCmdWithExitOnFailure(
     `az role assignment create --role "Managed Identity Operator" --assignee ${servicePrincipalClientId} --scope ${identity.id}`
   )
   // Allow the oracle identity to access the correct key vault
   await execCmdWithExitOnFailure(
-    `az keyvault set-policy --name ${keyVaultName(
-      index
-    )} --key-permissions {get,list,sign} --object-id ${identity.principalId} -g ${resourceGroup()}`
+    `az keyvault set-policy --name ${oracleIdentity.keyVaultName} --key-permissions {get,list,sign} --object-id ${identity.principalId} -g ${clusterConfig.resourceGroup}`
```

Comment:
Do we need to clean these up when an Oracle is moved or deleted?

Reply:
Great idea! Added that